pkgs/io_file/lib/src/vm_windows_file_system.dart

```diff
@@ -68,6 +68,18 @@ Exception _getError(int errorCode, String message, String path) {
 
 /// A [FileSystem] implementation for Windows systems.
 base class WindowsFileSystem extends FileSystem {
+  @override
+  void createDirectory(String path) => using((arena) {
+    // Calling `GetLastError` for the first time causes the `GetLastError`
+    // symbol to be loaded, which resets `GetLastError`. So make a harmless
+    // call before the value is needed.
+    win32.GetLastError();
```

Comment:
We may want to either:
- create a small 'primeLastError()' method in this library which does the same thing; this would let us avoid copying the 'why are we doing this' comment many times, or
- have a lastError() wrapper which accepts a closure; the wrapper can handle last error stuff, and the closure can do the actual work. this pattern may be less clear though

Reply:
Great idea! I'll go with the first idea, which will make it easy to remove when I fix this in the SDK.